File: jump-game-ii/jump-game-ii.cpp
#define MAX(a,b) ((a) > (b) ? (a) : (b))
class Solution {
public:
    int jump(vector<int>& nums) {
        if(nums.size() <= 1) {
            return 0;
        }
        int minJumpCnt = 1;
        int curMaxPos = nums[0];
        int nextMaxPos = nums[0];
        for(int pos = 1; pos < nums.size()-1; pos++) {
            nextMaxPos = MAX(nextMaxPos, pos + nums[pos]);
            if(pos == curMaxPos) {
                minJumpCnt++;
                curMaxPos = nextMaxPos;
            }
        }
        return minJumpCnt;
        
    }
};

File: jump-game-ii/jump-game-ii.ts
function jump(nums: number[]): number {
    let nextMaxPos = 0;
    let curMaxPos = 0;
    let minJumpCnt = 0;
    nums.forEach( (num, pos) => {
        nextMaxPos = Math.max(nextMaxPos, num + pos);
        if(curMaxPos === pos && pos < nums.length - 1) {
            curMaxPos = nextMaxPos;
            minJumpCnt++;
        }
    })
    return minJumpCnt;
};